src/corecontext.h

```diff
@@ -0,0 +1,28 @@
+// As shown here: https://github.com/mortennobel/QtOpenGL3.2Core
+
+#ifndef DISPLAZ_CORECONTEXT_H
+#define DISPLAZ_CORECONTEXT_H
+
+#include <QGLContext>
+#include <QGLFormat>
+
+#if defined(Q_OS_MAC)
+#if QT_VERSION < 0x050000 && QT_VERSION >= 0x040800 // if less than Qt 5.0.0
+void* select_3_2_mac_visual(GDHandle handle, int depthBufferSize, bool multiSample=false);
+#endif
+#endif
+
+struct CoreContext : public QGLContext {
+    CoreContext(const QGLFormat& format, QPaintDevice* device) : QGLContext(format,device) {}
+    CoreContext(const QGLFormat& format) : QGLContext(format) {}
+
+#if defined(Q_OS_MAC)
+#if QT_VERSION < 0x050000 && QT_VERSION >= 0x040800 // if less than Qt 5.0.0
+    virtual void* chooseMacVisual(GDHandle handle) {
+        return select_3_2_mac_visual(handle, this->format().depthBufferSize()); //TODO: check if we want to enable multiSampling
+    }
+#endif
+#endif
+};
+
+#endif //DISPLAZ_CORECONTEXT_H
```

Comment:
Eek, it's impressive that this works (not that I can test it) ;-)

Do you think people will want support for older Qt versions on OS X?  If not, I'd tentatively suggest we say it's version 5+ only and just delete the legacy support.

I'll certainly be using Qt5-only for the windows build as soon as I get the chance to fiddle with the build system.